lesson4/README.md

```diff
@@ -1,9 +1,93 @@
+    ```java
+    SharedPreferences sharedPref = getActivity().getSharedPreferences(
+            getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+    ```
+
+2. [`getPreferences()`](https://developer.android.com/reference/android/app/Activity.html#getPreferences(int)): Here you are pretty much doing the same thing except you don't need to mention the file name.
+    ```java
+    SharedPreferences sharedPref = getActivity().getPreferences(Context.MODE_PRIVATE);
+    ```
+
+So, whenever a session of the app gets created (inside `onCreate()`), you should read the shared preference file and when the app gets closed (inside `onStop()`), changes made (for ex. the backgroud color is changed to blue) should be written to the file.
+
+```java
+// To read
+SharedPreferences sharedPref = getActivity().getPreferences(Context.MODE_PRIVATE);
+// Default value for when the file is unavilable
+String defaultValue = getResources().getString(R.string.saved_background_default);
+String background = sharedPref.getString(getString(R.string.saved_background), defaultValue);
+
+// To write
+SharedPreferences sharedPref = getActivity().getPreferences(Context.MODE_PRIVATE);
+SharedPreferences.Editor editor = sharedPref.edit();
+editor.putString(getString(R.string.saved_background), currentBackground);
+editor.commit();
+```
+While writing, notice that we gave it the key (`getString(R.string.saved_background)`) and the corresponding value (`currentBackground`).
 
 ### SQL
 
-## App Life Cycle
\ No newline at end of file
+While implementing a SQLite database in Android, you almost always need to make a schema. The schema basically tells the database what columns are needed. This [page](https://developer.android.com/training/basics/data-storage/databases.html) has all the information you might need to add a SQLite database to your app. 
+
+## App Life Cycle
+
+![](https://developer.android.com/images/training/basics/basic-lifecycle.png)
+
+When you create an new project in Android Studio, you will see a lot of boilerplate. If you are new to Android dev, you probably wouldn't have gone beyond using `onCreate()` and `onDestroy()`. But as you advance, you will realize the need for other methods. Other that the create and destroy methods, we'll use `onPause()` and `onResume`.
+
+Read through each of the lessons listed in [this page](https://developer.android.com/training/basics/activity-lifecycle/index.html).
+
+## Homework
+
+1. Make a simple app that takes an RGB value from the user. The app sets the background color to that specified RGB value and also save the value in SharedPreferences. Use the correct methods of activity life cycle to figure out where to save and retrive those preferences. Test your app by pausing, resuming, quitting and restarting the app.
+2. TODO: Wait for Bill to figure out the HW for lesson 3. It will be some form of "Add SQL storage to the shopping list app"
```

Comment:
like i've said before, we need to discuss what the ongoing app that each lesson builds off on is